added tf_convenience_topics package

svbhat opened the pull request:
Added a package to listen to the tf-topics and publish state feedback.

nilsbore commented:
@svbhat see https://github.com/smarc-project/smarc_navigation/pull/54/checks?check_run_id=2085889977#step:3:852 . Do we need `cola2_msgs` for this package? Or is that a transitive dependency?

svbhat commented:
> @svbhat see https://github.com/smarc-project/smarc_navigation/pull/54/checks?check_run_id=2085889977#step:3:852 . Do we need `cola2_msgs` for this package? Or is that a transitive dependency?

I don't think we need it for the tf_listener. I had it before for the controllers repo. I have removed it now. Hopefully it passes the CI check now!

nilsbore commented on tf_convenience_topics/CMakeLists.txt:
```diff
@@ -0,0 +1,218 @@
+
+## Mark executables for installation
+## See http://docs.ros.org/melodic/api/catkin/html/howto/format1/building_executables.html
+# install(TARGETS ${PROJECT_NAME}_node
```
Please add install rules for all nodes and libraries!

nilsbore replied:
And launch files :)

svbhat replied:
Sorry for missing that! Could you please share an example I can follow? I followed the smarc_controllers repo and there were no install rules in the CMakelists.txt there.

nilsbore replied:
Check any of the released repos: https://github.com/smarc-project/rosinstall#package-status

nilsbore replied:
You can also check the commented install rules in your file

nilsbore commented:
Note that you still included the `cola2_msgs` DVL message: https://github.com/smarc-project/smarc_navigation/runs/2086211619#step:3:851 (the previous commit failed, see red cross next to it) . Don't know if you removed that in the last commit.

svbhat commented:
> Note that you still included the `cola2_msgs` DVL message: https://github.com/smarc-project/smarc_navigation/runs/2086211619#step:3:851 (the previous commit failed, see red cross next to it) . Don't know if you removed that in the last commit.

Oh yes! Thanks for pointing that out. The DVL message is of type 'cola2_msgs' and I look at it to get the altitude. If I want to include that dependency, should I also point to the git repository of cola2_msgs in the package.rosinstall file?

svbhat commented:
Now it passes all checks. I've added cola2_msgs to the rosinstall file, and updated the release_packages.yaml file to include the new package.

nilsbore commented:
> Now it passes all checks. I've added cola2_msgs to the rosinstall file, and updated the release_packages.yaml file to include the new package.

Ok, good work. I didn't realize you were actually using the DVL. Actually we've switched to `smarc_msgs/DVL`(see https://github.com/smarc-project/smarc_msgs/blob/noetic-devel/msg/DVL.msg , it's the same as in `cola2_msgs`). Can you please switch to that and remove the `cola2_msgs` dep again? That should be the last thing I hope :)

svbhat commented:
> > Now it passes all checks. I've added cola2_msgs to the rosinstall file, and updated the release_packages.yaml file to include the new package.
> 
> Ok, good work. I didn't realize you were actually using the DVL. Actually we've switched to `smarc_msgs/DVL`(see https://github.com/smarc-project/smarc_msgs/blob/noetic-devel/msg/DVL.msg , it's the same as in `cola2_msgs`). Can you please switch to that and remove the `cola2_msgs` dep again? That should be the last thing I hope :)

Done, updated with smarc_msgs/DVL now.
